device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -158,8 +158,10 @@ public void open(Queue<DeviceClientConfig> deviceClientConfigs) throws Transport
             }
             catch (InterruptedException e)
             {
-                executorServicesCleanup();
```

Comment:
I am assuming we only want to clean up on close?

Reply:
The ```close``` method already calls ```executorServicesCleanup```, so I'm just delaying it until after the reactor has stopped